## Compter les étudiants EDI dont les parents ont fait des études ou non :

In [2]:
import pandas as pd

# Charger le fichier Excel
file_path = 'data/Données VI [GEN + MIG] + VD [SENSE + IND + INTER].xlsx'
df = pd.read_excel(file_path)

# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Compter le nombre d'individus dans chaque groupe
count_first_gen_edi = first_gen_edi.shape[0]
count_second_gen_edi = second_gen_edi.shape[0]

# Afficher les résultats
print(f"Nombre d'individus '1ST GEN' et 'EDI' : {count_first_gen_edi}")
print(f"Nombre d'individus '2ND GEN' et 'EDI' : {count_second_gen_edi}")


Nombre d'individus '1ST GEN' et 'EDI' : 61
Nombre d'individus '2ND GEN' et 'EDI' : 53


In [18]:
import pandas as pd
import statsmodels.api as sm

# Charger le fichier Excel
file_path = 'data/Données VI [GEN + MIG] + VD [SENSE + IND + INTER].xlsx'
df = pd.read_excel(file_path)

# Filtrer les données pour "1ST GEN" dans la colonne B et "EDI" dans la colonne C
filtered_df = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Sélectionner les colonnes pertinentes
# Colonne I : MOY SENSE [VD]
y = filtered_df['MOY SENSE [VD]'].astype(float)  # Convertir en float si nécessaire

# Créer une variable indépendante fictive pour l'intercept (une colonne de 1)
x = pd.DataFrame({'Intercept': [1] * len(y)}, index=y.index)  # Aligner les indices avec y
x = sm.add_constant(x)  # Ajouter une constante

# Ajuster le modèle de régression linéaire
model = sm.OLS(y, x).fit()

# Afficher le résumé des résultats de la régression
print(model.summary())


                            OLS Regression Results                            
Dep. Variable:         MOY SENSE [VD]   R-squared:                      -0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                       nan
Date:                Fri, 24 May 2024   Prob (F-statistic):                nan
Time:                        21:06:14   Log-Likelihood:                -47.266
No. Observations:                  61   AIC:                             96.53
Df Residuals:                      60   BIC:                             98.64
Df Model:                           0                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      4.3714      0.068     64.479      0.0

## Code pour comparer le sentiment d'appartenancedes étudiants d'origine imigrée selon le niveau d'étude des parents :

In [20]:
import pandas as pd
from scipy import stats

file_path = 'data/Données VI [GEN + MIG] + VD [SENSE + IND + INTER].xlsx'
df = pd.read_excel(file_path)

first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

t_stat, p_value = stats.ttest_ind(first_gen_edi, second_gen_edi, nan_policy='omit')

print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 0.06427219588276172
P-value: 0.9488680663294495
Il n'y a pas de différence statistiquement significative entre les deux groupes.


## Ce code effectue les étapes suivantes :
Charger les données depuis le fichier Excel.
Filtrer les données pour obtenir les valeurs de MOY SENSE [VD] des individus classés comme "1ST GEN" et "EDI".
Filtrer les données pour obtenir les valeurs de MOY SENSE [VD] des individus classés comme "2ND GEN" et "EDI".
Effectuer un test t pour comparer les moyennes des deux groupes.
Afficher les résultats du test t et une interprétation de ces résultats.


Résultats du Test T

T-statistic : 0.0643

P-value : 0.9489

Interprétation: 
La valeur de p (0.9489) est bien supérieure au seuil de signification couramment utilisé de 0.05. Par conséquent, nous n'avons pas suffisamment de preuves pour rejeter l'hypothèse nulle, ce qui signifie qu'il n'y a pas de différence statistiquement significative entre les valeurs moyennes de MOY SENSE [VD] des individus classés comme "1ST GEN" et "EDI" et ceux classés comme "2ND GEN" et "EDI".

En d'autres termes, les moyennes de MOY SENSE [VD] pour ces deux groupes sont très similaires selon les données disponibles.

## Moyenne sentiment d'appartenance pour chaque groupe EDI 1e génération VS 2e génération: 

In [15]:
import pandas as pd

# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

# Calculer les moyennes pour chaque groupe
mean_first_gen_edi = first_gen_edi.mean()
mean_second_gen_edi = second_gen_edi.mean()

# Afficher les moyennes
print(f"Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : {mean_first_gen_edi}")
print(f"Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : {mean_second_gen_edi}")


Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : 4.371374527112231
Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : 4.365021770682149


# Comparaison des moyennes de sentiment d'appartenance entre étudiants de parents qui ont fait des études VS sans études 

In [16]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "1ST GEN"
first_gen = df[df['GEN (capital culturel) [VI]'] == '1ST GEN']['MOY SENSE [VD]'].astype(float)

# Filtrer les données pour "2ND GEN"
second_gen = df[df['GEN (capital culturel) [VI]'] == '2ND GEN']['MOY SENSE [VD]'].astype(float)

# Effectuer le test t pour deux échantillons indépendants
t_stat, p_value = stats.ttest_ind(first_gen, second_gen, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_first_gen = first_gen.mean()
mean_second_gen = second_gen.mean()

print(f"Moyenne de MOY SENSE [VD] pour 1ST GEN : {mean_first_gen}")
print(f"Moyenne de MOY SENSE [VD] pour 2ND GEN : {mean_second_gen}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 0.4746915437296905
P-value: 0.635321779756749
Moyenne de MOY SENSE [VD] pour 1ST GEN : 4.297670639219935
Moyenne de MOY SENSE [VD] pour 2ND GEN : 4.269027269027273
Il n'y a pas de différence statistiquement significative entre les deux groupes.


In [19]:
import pandas as pd
from scipy import stats


# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY IND [VD]'].astype(float)

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY IND [VD]'].astype(float)

# Effectuer le test t pour deux échantillons indépendants
t_stat, p_value = stats.ttest_ind(first_gen_edi, second_gen_edi, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_first_gen_edi = first_gen_edi.mean()
mean_second_gen_edi = second_gen_edi.mean()

print(f"Moyenne de MOY IND [VD] pour 1ST GEN et EDI : {mean_first_gen_edi}")
print(f"Moyenne de MOY IND [VD] pour 2ND GEN et EDI : {mean_second_gen_edi}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 0.35722870007135
P-value: 0.7215928284176576
Moyenne de MOY IND [VD] pour 1ST GEN et EDI : 5.992974238875879
Moyenne de MOY IND [VD] pour 2ND GEN et EDI : 5.913746630727763
Il n'y a pas de différence statistiquement significative entre les deux groupes.


In [20]:
import pandas as pd
from scipy import stats


# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY INTER [VD]'].astype(float)

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY INTER [VD]'].astype(float)

# Effectuer le test t pour deux échantillons indépendants
t_stat, p_value = stats.ttest_ind(first_gen_edi, second_gen_edi, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_first_gen_edi = first_gen_edi.mean()
mean_second_gen_edi = second_gen_edi.mean()

print(f"Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : {mean_first_gen_edi}")
print(f"Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : {mean_second_gen_edi}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 2.313860779048787
P-value: 0.02249978812932491
Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : 5.819672131147539
Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : 5.216981132075472
Il y a une différence statistiquement significative entre les deux groupes.


In [21]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "1ST GEN"
first_gen = df[df['GEN (capital culturel) [VI]'] == '1ST GEN']['MOY IND [VD]'].astype(float)

# Filtrer les données pour "2ND GEN"
second_gen = df[df['GEN (capital culturel) [VI]'] == '2ND GEN']['MOY IND [VD]'].astype(float)

# Effectuer le test t pour deux échantillons indépendants
t_stat, p_value = stats.ttest_ind(first_gen, second_gen, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_first_gen = first_gen.mean()
mean_second_gen = second_gen.mean()

print(f"Moyenne de MOY IND [VD] pour 1ST GEN : {mean_first_gen}")
print(f"Moyenne de MOY IND [VD] pour 2ND GEN : {mean_second_gen}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 0.741709275375226
P-value: 0.45879263203860976
Moyenne de MOY IND [VD] pour 1ST GEN : 5.764587525150904
Moyenne de MOY IND [VD] pour 2ND GEN : 5.6689342403628125
Il n'y a pas de différence statistiquement significative entre les deux groupes.


In [22]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "1ST GEN"
first_gen = df[df['GEN (capital culturel) [VI]'] == '1ST GEN']['MOY INTER [VD]'].astype(float)

# Filtrer les données pour "2ND GEN"
second_gen = df[df['GEN (capital culturel) [VI]'] == '2ND GEN']['MOY INTER [VD]'].astype(float)

# Effectuer le test t pour deux échantillons indépendants
t_stat, p_value = stats.ttest_ind(first_gen, second_gen, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_first_gen = first_gen.mean()
mean_second_gen = second_gen.mean()

print(f"Moyenne de MOY INTER [VD] pour 1ST GEN : {mean_first_gen}")
print(f"Moyenne de MOY INTER [VD] pour 2ND GEN : {mean_second_gen}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes.")


T-statistic: 4.87033312922056
P-value: 1.7338324686403362e-06
Moyenne de MOY INTER [VD] pour 1ST GEN : 5.34154929577465
Moyenne de MOY INTER [VD] pour 2ND GEN : 4.495590828924163
Il y a une différence statistiquement significative entre les deux groupes.


In [24]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "1ST GEN" et "EDI" pour MOY INTER [VD]
first_gen_edi_inter = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY INTER [VD]'].astype(float)

# Filtrer les données pour "2ND GEN" et "EDI" pour MOY INTER [VD]
second_gen_edi_inter = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY INTER [VD]'].astype(float)

# Filtrer les données pour "1ST GEN" et "EDI" pour MOY SENSE [VD]
first_gen_edi_sense = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

# Filtrer les données pour "2ND GEN" et "EDI" pour MOY SENSE [VD]
second_gen_edi_sense = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]['MOY SENSE [VD]'].astype(float)

# Effectuer le test t pour MOY INTER [VD]
t_stat_inter, p_value_inter = stats.ttest_ind(first_gen_edi_inter, second_gen_edi_inter, nan_policy='omit')

# Effectuer le test t pour MOY SENSE [VD]
t_stat_sense, p_value_sense = stats.ttest_ind(first_gen_edi_sense, second_gen_edi_sense, nan_policy='omit')

# Afficher les résultats pour MOY INTER [VD]
print("Comparaison des moyennes de MOY INTER [VD] :")
print(f"T-statistic: {t_stat_inter}")
print(f"P-value: {p_value_inter}")

# Calculer et afficher les moyennes pour chaque groupe pour MOY INTER [VD]
mean_first_gen_edi_inter = first_gen_edi_inter.mean()
mean_second_gen_edi_inter = second_gen_edi_inter.mean()

print(f"Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : {mean_first_gen_edi_inter}")
print(f"Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : {mean_second_gen_edi_inter}")

# Interprétation des résultats pour MOY INTER [VD]
if p_value_inter < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes pour MOY INTER [VD].")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes pour MOY INTER [VD].")

# Afficher les résultats pour MOY SENSE [VD]
print("\nComparaison des moyennes de MOY SENSE [VD] :")
print(f"T-statistic: {t_stat_sense}")
print(f"P-value: {p_value_sense}")

# Calculer et afficher les moyennes pour chaque groupe pour MOY SENSE [VD]
mean_first_gen_edi_sense = first_gen_edi_sense.mean()
mean_second_gen_edi_sense = second_gen_edi_sense.mean()

print(f"Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : {mean_first_gen_edi_sense}")
print(f"Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : {mean_second_gen_edi_sense}")

# Interprétation des résultats pour MOY SENSE [VD]
if p_value_sense < 0.05:
    print("Il y a une différence statistiquement significative entre les deux groupes pour MOY SENSE [VD].")
else:
    print("Il n'y a pas de différence statistiquement significative entre les deux groupes pour MOY SENSE [VD].")


Comparaison des moyennes de MOY INTER [VD] :
T-statistic: 2.313860779048787
P-value: 0.02249978812932491
Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : 5.819672131147539
Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : 5.216981132075472
Il y a une différence statistiquement significative entre les deux groupes pour MOY INTER [VD].

Comparaison des moyennes de MOY SENSE [VD] :
T-statistic: 0.06427219588276172
P-value: 0.9488680663294495
Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : 4.371374527112231
Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : 4.365021770682149
Il n'y a pas de différence statistiquement significative entre les deux groupes pour MOY SENSE [VD].


## Comparaison motivations INTER vs Sens of B, pour EDI 1GEN : 

In [26]:
import pandas as pd
from scipy import stats


# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Extraire les colonnes MOY INTER [VD] et MOY SENSE [VD]
inter_values = first_gen_edi['MOY INTER [VD]'].astype(float)
sense_values = first_gen_edi['MOY SENSE [VD]'].astype(float)

# Effectuer le test t pour échantillons appariés
t_stat, p_value = stats.ttest_rel(inter_values, sense_values, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_inter = inter_values.mean()
mean_sense = sense_values.mean()

print(f"Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : {mean_inter}")
print(f"Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : {mean_sense}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 1ST GEN et EDI.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 1ST GEN et EDI.")


T-statistic: 10.013669329620646
P-value: 2.029899157953692e-14
Moyenne de MOY INTER [VD] pour 1ST GEN et EDI : 5.819672131147539
Moyenne de MOY SENSE [VD] pour 1ST GEN et EDI : 4.371374527112231
Il y a une différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 1ST GEN et EDI.


## Comparaison motivations INTER vs Sens of B, pour EDI 2nd GEN:

In [27]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Extraire les colonnes MOY INTER [VD] et MOY SENSE [VD]
inter_values = second_gen_edi['MOY INTER [VD]'].astype(float)
sense_values = second_gen_edi['MOY SENSE [VD]'].astype(float)

# Effectuer le test t pour échantillons appariés
t_stat, p_value = stats.ttest_rel(inter_values, sense_values, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Calculer et afficher les moyennes pour chaque groupe
mean_inter = inter_values.mean()
mean_sense = sense_values.mean()

print(f"Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : {mean_inter}")
print(f"Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : {mean_sense}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 2ND GEN et EDI.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 2ND GEN et EDI.")


T-statistic: 3.8407014630073113
P-value: 0.00033495331707233587
Moyenne de MOY INTER [VD] pour 2ND GEN et EDI : 5.216981132075472
Moyenne de MOY SENSE [VD] pour 2ND GEN et EDI : 4.365021770682149
Il y a une différence statistiquement significative entre les moyennes de MOY INTER [VD] et MOY SENSE [VD] pour les étudiants 2ND GEN et EDI.


In [28]:
import pandas as pd
from scipy import stats

# Filtrer les données pour "1ST GEN" et "EDI"
first_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '1ST GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Filtrer les données pour "2ND GEN" et "EDI"
second_gen_edi = df[(df['GEN (capital culturel) [VI]'] == '2ND GEN') & (df['MIG (ascendance migratoire) [VI]'] == 'EDI')]

# Extraire les colonnes MOY INTER [VD] et MOY SENSE [VD] pour les deux groupes
inter_values_1st_gen = first_gen_edi['MOY INTER [VD]'].astype(float)
sense_values_1st_gen = first_gen_edi['MOY SENSE [VD]'].astype(float)

inter_values_2nd_gen = second_gen_edi['MOY INTER [VD]'].astype(float)
sense_values_2nd_gen = second_gen_edi['MOY SENSE [VD]'].astype(float)

# Calculer les différences de moyennes pour chaque groupe
difference_1st_gen = inter_values_1st_gen - sense_values_1st_gen
difference_2nd_gen = inter_values_2nd_gen - sense_values_2nd_gen

# Calculer et afficher les moyennes des différences pour chaque groupe
mean_difference_1st_gen = difference_1st_gen.mean()
mean_difference_2nd_gen = difference_2nd_gen.mean()

print(f"Différence moyenne de MOY INTER - MOY SENSE pour 1ST GEN et EDI : {mean_difference_1st_gen}")
print(f"Différence moyenne de MOY INTER - MOY SENSE pour 2ND GEN et EDI : {mean_difference_2nd_gen}")

# Comparer les deux différences de moyennes
if mean_difference_1st_gen > mean_difference_2nd_gen:
    print("La différence moyenne de MOY INTER - MOY SENSE est plus grande pour le groupe 1ST GEN et EDI.")
elif mean_difference_1st_gen < mean_difference_2nd_gen:
    print("La différence moyenne de MOY INTER - MOY SENSE est plus grande pour le groupe 2ND GEN et EDI.")
else:
    print("Les différences moyennes de MOY INTER - MOY SENSE sont égales pour les deux groupes.")

# Effectuer le test t pour comparer les différences de moyennes entre les deux groupes
t_stat, p_value = stats.ttest_ind(difference_1st_gen, difference_2nd_gen, nan_policy='omit')

# Afficher les résultats
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interprétation des résultats
if p_value < 0.05:
    print("Il y a une différence statistiquement significative entre les différences de moyennes de MOY INTER et MOY SENSE entre les groupes 1ST GEN et 2ND GEN pour les étudiants EDI.")
else:
    print("Il n'y a pas de différence statistiquement significative entre les différences de moyennes de MOY INTER et MOY SENSE entre les groupes 1ST GEN et 2ND GEN pour les étudiants EDI.")


Différence moyenne de MOY INTER - MOY SENSE pour 1ST GEN et EDI : 1.4482976040353086
Différence moyenne de MOY INTER - MOY SENSE pour 2ND GEN et EDI : 0.8519593613933235
La différence moyenne de MOY INTER - MOY SENSE est plus grande pour le groupe 1ST GEN et EDI.
T-statistic: 2.3073171126351677
P-value: 0.022874895855250203
Il y a une différence statistiquement significative entre les différences de moyennes de MOY INTER et MOY SENSE entre les groupes 1ST GEN et 2ND GEN pour les étudiants EDI.
